alphagov/paas-cf
# [#118836855] Load CIDRs of tenants from file per environment and Whitelist Bit Zesty IP address

[#118836855 Whitelist Bit Zesty IP address](https://www.pivotaltracker.com/story/show/118836855)

What
----

As we currently restrict access to Aviation House for internet facing system components, we need to  allow access to Bit Zesty before we implement a more secure mechanism to access the platform.

We want to implement it in a way that will:
 * Allow us to easily add more tenants
 * Allow different IPs in prod than dev/staging (3rd party tenants will access prod)

Context
-------

To implement this, we allow define in file tenant CIDRs with access to API+web by using terraform interpolation.

Terraform will read from a file defined by the variable ${var.tenant_cidrs_file}, one CIDR per line and allowing comments.

Prod will be the environment access for the actual tenants, so for dev and staging it will load the file
terraform/tenant-cidrs-default.txt, and prod will load terraform/tenant-cidrs-prod.txt as defined in  erraform/prod.tfvars.

We move the office CIDRs to these files too, but keep the variable office_cirds for more specific and operator related security groups.

The variable ${var.web_access_cidrs} is now empty, but will be updated in prod to allow access from the world.

How to test it
--------------

 1. Deploy the environment and confirm that the office IPs are still being properly added to the SG for web and api access.
 2. Optionally add other IP to `terraform/tenant-cidrs-default.txt`
 3. Optionally override the variable `$var.tenant_cidrs_file}` to point to prod and see that it adds the Bit Zesty IP to the SG.

Who?
----

Anyone but @keymon or @combor
